mypy/stubtest.py

```diff
@@ -1256,6 +1256,19 @@ def verify_paramspecexpr(
         return
 
 
+def _is_django_cached_property(runtime: Any) -> bool:  # pragma: no cover
+    # This is a special case for
+    # https://docs.djangoproject.com/en/5.2/ref/utils/#django.utils.functional.cached_property
+    # This is needed in `django-stubs` project:
+    # https://github.com/typeddjango/django-stubs
+    if repr(type(runtime)) != "<class 'django.utils.functional.cached_property'>":
```

Comment:
What do you think of using `type(runtime).__name__ == "cached_property"` as a heuristic? That way we're not as specific to one user.

Also out of curiosity, why can't django use `@functools.cached_property` directly?

Reply:
> Also out of curiosity, why can't django use @functools.cached_property directly?

No idea 🤷

> What do you think of using type(runtime).__name__ == "cached_property" as a heuristic? That way we're not as specific to one user.

I like your idea better, because it does not call `__repr__`, which might have side-effects. Of course, `.__name__` might also trigger some very complex cases, but they should be extrimelly rare.

Reply:
> Why can't django use `@functools.cached_property` directly?

Django predates python 3.8 by many years, and probably there's little to no benefit from removing their in-house version. A lot (really a lot) of users import `cached_property` from Django, so they'll have to maintain an alias anyway.

Reply:
Yup. Django's version long predates Python's. Also, when it was originally introduced in Python 3.8, the implementation had a lock that caused a lot of problems which was only only removed in Python 3.12. As Django 6.0 will target Python 3.12+ we can probably look at deprecating Django's version, or at least making it an alias of the standard library version.